Ecommerce/app/controllers/categories_controller.rb

```diff
@@ -1,7 +1,7 @@
 class CategoriesController < ApplicationController
   def index
-    @category = params[:category]
-    @items = @category.present? ? Item.where(category: @category) : Item.all
+    @q = Item.ransack(params[:q])
```

Comment:
in categories controller why we are applying ransack on item?

Reply:
When I created the Items controller, it wasn't being used, so you told me to delete it. Then, I displayed the items in the Categories controller, that is why I used Ransack there.

Reply:
But now is the time to use the items controller,  you can not write the items controller index method in categories controller index method.
Whenever I told you to remove or add something it has some purpose. But the purpose does not change the rules for future developement,